src/mumble/AudioWizard.cpp

```diff
@@ -355,14 +355,21 @@ int AudioWizard::nextId() const {
 }
 
 void AudioWizard::playChord() {
+	if (!m_chordEnabled) {
+		return;
+	}
```

Comment:
Why do we need this helper flag? It seems like if chord is not enabled, this function should simply not be called in the first place :thinking:

Reply:
Excellent question. When I tried this with the original code flow the "This is an example of Speex" sample kept looping when the wizard was closed. This is because the ``restartAudio`` method, which is called by accept and reject, starts the sample again after the audio system is reinitialized.
Why on earth this ever worked before I don't know. Must be that the old logic was broken because of the race conditions we discovered here. Or maybe I missed something... Anyway the flag seemed like the most elegant solution to me at the time of writing.

Reply:
Passing a true or false to each call of ``restartAudio`` as parameter would also work, if you like that more...

Reply:
Yeah, I think a boolean argument to `restartAudio` would be better as this makes the flow of events much more direct. Aka: it's not needed to trace where the chord-flag variable is set such that subsequent calls to `restartAudio` won't cause the chord to play (indirectly).
Instead, you will see a call to `restartAudio` and immediately in that call you see whether a restart of the chord is requested or not. To me this seems simpler :)